src/universal/Atmosphere.js

```diff
@@ -32,31 +70,60 @@ export default class Atmosphere extends Environment {
 
     // now atmosphere
     this.authToken = undefined;
-    this.socket = undefined;
+    this.subscriptionClient = undefined;
     this.networks = {
       http: this._network,
       socket: Network.create(this.fetchWS, this.socketSubscribe)
     };
-
-    this.opIdIndex = 1;
     this.querySubscriptions = [];
     this.subscriptions = {};
+    this.useSubscriptions = true;
   }
 
-  use = (networkName) => {
-    // TODO build this out when we start hitting github directly
-    return this.environments[networkName];
-  };
+  makeSubscriptionClient() {
+    if (!this.subscriptionClient) {
+      this.subscriptionClient = new Promise((resolve, reject) => {
```

Comment:
Storing the promise on `this` feels messy to me. I see two approaches to making this cleaner:

- Just return the promise to `subscriptionClient`
- Set `this.subscriptionClient` once the socket is open instead of setting `this.subscriptionClient` to the promise.

Reply:
> Set this.subscriptionClient once the socket is open

Can't, the app calls ensureSubscriptionClient before the onopen fires.
if this were typescript, this.subscriptionClient would be private & the getter would be ensureSubscriptionClient. components call ensureSubscriptionClient before onopen occurs, and once it does resolve, we only want 1 set of callbacks put on the socket. Block your sockets & play around with it to see what I mean. Open to PRs if you can figure out a better way!